Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs

```diff
@@ -59,23 +55,25 @@ public QuestionRepository(TrappistDbContext dbContext, UserManager<ApplicationUs
         }
 
         /// <summary>
-        /// Add new code snippet Question to the database
+        /// Adds new code snippet question to the Database
         /// </summary>
-        /// <param name="questionAC">Question data transfer object</param>
-        public async Task AddCodeSnippetQuestionAsync(QuestionAC questionAC)
+        /// <param name="questionAC">QuestionAC class object</param>
+        /// <param name="userId">Id of logged in user</param>
+        public async Task AddCodeSnippetQuestionAsync(QuestionAC questionAC, string userId)
         {
-            var codeSnippetQuestion = questionAC.CodeSnippetQuestion;
+            var codeSnippetQuestion = Mapper.Map< CodeSnippetQuestionAC, CodeSnippetQuestion>(questionAC.CodeSnippetQuestion);
             var question = Mapper.Map<QuestionDetailAC, Question>(questionAC.Question);
+            question.CreatedByUserId = userId;
 
             using (var transaction = _dbContext.Database.BeginTransaction())
             {
                 //Add common question details
-                await _dbContext.Question.AddAsync(question);
+                _dbContext.Question.Add(question);
```

Comment:
here you have used Add and in below statement you have used AddAsync? Can you say why?